language/tools/move-mv-llvm-compiler/tests/rbpf-tests/entry-point04.json

```diff
@@ -0,0 +1,17 @@
+{
+    "program_id": "DozgQiYtGbdyniV2T74xMdmjZJvYDzoRFFqw7UR5MwPK",
+    "accounts": [
+        {
+            "key": "524HMdYYBy6TAn4dK5vCcjiTmT2sxV6Xoue5EXrz22Ca",
+            "owner": "BPFLoaderUpgradeab1e11111111111111111111111",
+            "is_signer": false,
+            "is_writable": true,
+            "lamports": 1000,
+            "data": [
+                59, 180, 10, 27, 53, 174, 68, 250, 182, 197, 64, 125, 20, 13, 245, 130, 63, 209, 147, 189, 65, 82, 20, 53, 37, 71, 8, 125, 94, 87, 32, 27,
+                2, 3, 5, 7, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 11, 13, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173,
+                17, 19, 23, 29, 31, 37, 41, 43]
+        }
```

Comment:
I assume the creator of the data array will insert padding to ensure proper alignment. For example, consider above modified with an additional `u8` parameter-- instead of entry point data `&signer, address, u64` we have entry point with `u8, &signer, address, u64`. Does the array get 1 byte + 7 bytes padding in the first slot? Otherwise, I think we can get misaligned reads for the subsequent parameters. I don't remember if RBPF VM aborts the process or just returns bad data in that case. Or instead of adding padding we could avoid raw `load` and `store` and write runtime load and store routines that support misalignment (but slower operation).

Reply:
This is a very good point.  I think handling this depends on how the client side API will serialize the parameter values. If it pads the values to keep everything properly aligned, we should expect the padding. If it packs everything without gaps, we would have to handle unaligned reads. Unlikely users will put anything directly into instruction_data (if they do, they can't expect it to work correctly), but we should extend the client API to prepare arguments to be passed to Move entry functions and make a decision about alignment and padding at that point.